plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -183,20 +185,42 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 		return;
 	}
 
+	is_host = fu_plugin_thunderbolt_is_host (device);
+
 	vid = fu_plugin_thunderbolt_udev_get_id (device, "vendor", &error);
 	if (vid == 0x0) {
 		g_warning ("failed to get Vendor ID: %s", error->message);
-		return;
+		if (is_host)
+			is_safemode = TRUE;
```

Comment:
I'm not sure this is a safe assumption. The read error can be from other things too. The only guaranteed way to identify safe-mode is by reading `nvm_version` and looking for `ENODATA` error. https://github.com/torvalds/linux/blob/master/Documentation/admin-guide/thunderbolt.rst#upgrading-nvm-when-host-controller-is-in-safe-mode

My suggestion is to take the version read up here, set `is_safemode` if it fails with `ENODATA` and then decide based on it if vid/did should be read from TBT sysfs attributes or from Dell.